src/conn/conn_ckpt.c

```diff
@@ -78,14 +94,20 @@ __ckpt_server(void *arg)
 	conn = S2C(session);
 	wt_session = (WT_SESSION *)session;
 
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
-	    F_ISSET(conn, WT_CONN_SERVER_CHECKPOINT)) {
+	for (;;) {
+		if (!__ckpt_server_run_chk(session))
+			break;
```

Comment:
Why do we need to check before calling `__wt_cond_wait`?  It's going to check before going to sleep...

I don't mind the `while(__wt_server_wait_loop(...))` construct (though I don't like `_and_` function names), since we're repeating the same logic in lots of places.  What's your objection @keithbostic?